README.md

```diff
@@ -18,25 +20,185 @@ $ npm install --save pipage
+```
+
+### Removing streams
+
+```js
+// Streams can be shifted off the beginning,
+var firstStream = pipeline.shift()
 
-if( bmap ) {
-  pipeline.splice( pipeline.indexOf( checksum ), 1 )
-  pipeline.append( new BlockMap.FilterStream( bmap ) )
+// or popped off the end,
+var lastStream = pipeline.pop()
+
+// spliced out at a specific index (remove 3 from index 2),
+var removedStreams = pipeline.splice( 2, 3 )
+
+// or removed by reference:
+pipeline.remove( stream )
+```
+
+### Selecting streams
+
+```js
+// Get a stream at a specific index in the pipeline:
+var stream = pipeline.get( 2 )
+var lastStream = pipeline.get( -1 )
+
+// Find a stream in the pipeline:
+var index = pipeline.indexOf( stream )
+var lastIndex = pipeline.lastIndexOf( stream )
+```
+
+### Example
+
+```js
+var Pipage = require('pipage')
+var path = require('path')
+var zlib = require('zlib')
+var unbzip2 = require('unbzip2-stream')
+var xz = require('xz')
+
+var pipeline = new Pipage()
+
+switch( path.extname( filename ) ) {
+  case '.gz':
+    pipeline.prepend( zlib.createUnzip() );
+    break
+  case '.bz':
+  case '.bz2':
+    pipeline.prepend( unbzip2() );
+    break
+  case '.xz':
+    pipeline.prepend( new xz.Decompressor() )
+    break
 }
 
-new BlockReadStream( '/path/to/resin-os.img' )
+fs.createReadStream( filename )
   .pipe( pipeline )
-  .pipe( new BlockWriteStream( '/dev/rdisk2' ) )
-  .once( 'finish', function() {
-    // happy face
-  })
+  .pipe( fs.createWriteStream( destination ) )
+```
+
+### Events
+
+#### Error handling
+
+Errors from streams within the pipeline are listened to,
+and re-emitted on the pipeline itself, with an additional `.stream` property
+being set on the `error` object, which is the stream that emitted it:
+
+```js
+pipeline.on( 'error', function( error ) {
+  // This error originated from the pipeline's internal stream
+  // available as `error.stream`, not from the pipeline itself
+  if( error.stream ) {
+    // ...
+  } else {
+    // This error came from the pipeline itself
+  }
+})
+```
+
+#### Binding to contained stream's events
+
+```js
+var stream = pipeline.get(-1)
+
+// Re-emit a stream's events on the pipeline:
+pipeline.bind( stream, 'eventname' )
+pipeline.bind( stream, [ 'someevent', 'otherevent' ])
+
+// Remove re-emission of a stream's event:
+pipeline.unbind( stream, 'eventname' )
+
+// Stop re-emission of all of stream's events on the pipeline:
+pipeline.unbindAll( stream )
+```
+
+##### Example
+
+```js
+var Pipage = require('pipage')
+
+// Let's say we have a stream which emits an event
+// we want to capture without having to get a reference to
+// that particular stream:
+module.exports = function createPipeline() {
+
+  var checksumStream = createChecksumStream( 'sha256', 'md5' )
+  var pipeline = new Pipage([ checksumStream ])
+
+  // This will cause the pipeline to re-emit
+  // the 'checksums' event from the `checksumStream`
+  pipeline.bind( checksumStream, 'checksums' )
```

Comment:
I'm not suggesting you change this - I'm merely asking out of curiosity... but could the above three lines theoretically be reduced to just:
```js
var pipeline = new Pipage([ createChecksumStream( 'sha256', 'md5' ) ])
pipeline.bind( pipeline.get(0), 'checksums' )
```
?